react-sdk/src/model/component-metadata.ts

```diff
@@ -37,26 +48,61 @@ export type TamboToolRegistry = Record<string, TamboTool>;
  *
  * Do not export this type from the SDK.
  */
-export type JSONSchemaLite = ReturnType<typeof zodToJsonSchema> & {
+export type JSONSchemaLite = JSONSchema7 & {
   description?: string;
 };
 
-export interface TamboTool<
-  Args extends z.ZodTuple<any, any> = z.ZodTuple<any, any>,
-  Returns extends z.ZodTypeAny = z.ZodTypeAny,
-> {
+/**
+ * A tool that can be registered with Tambo and invoked by the AI.
+ *
+ * Tools allow the AI to perform actions and retrieve information. Each tool has:
+ * - A unique name for identification
+ * - A description that helps the AI understand when to use it
+ * - A function that executes the tool's logic
+ * - A schema describing the tool's parameters (Standard Schema or JSON Schema)
+ * @example
+ * ```typescript
+ * import { z } from "zod/v4";
+ *
+ * const weatherTool: TamboTool = {
+ *   name: "getWeather",
+ *   description: "Get current weather for a location",
+ *   tool: (location: string) => fetchWeather(location),
+ *   toolSchema: z.function({
+ *    input: z.string(),
+ *    output: z.object({
+ *      temperature: z.number(),
+ *      conditions: z.string()
+ *     })
+ *   })
+ * };
+ * ```
+ */
+export interface TamboTool {
```

Comment:
sad to see that there's no easy way to keep `Args` and `Returns` - I assume this just gets incredibly hard when you're dealing with a combination of zod and StandardSchema?

Reply:
well what already worked was you could assign the tool, and then pass in schema, and the type system would complain if things didn't line up:
```
{
  tool: (message: string) => { console.log("I got a message: ", message); return "Hello!" },
  toolSchema: z.function().args(z.number()).returns(z.number())
 }
 ```
and you'd get a type error because the args / return values do not match

Reply:
I see, yeah I wasn't getting those because it wasn't specifying the type args. I'm going to try something to fix this.